source/simulator/assembly.cc

```diff
@@ -895,6 +915,36 @@ namespace aspect
                                                         advection_field);
       }
 
+    if (use_smoothing == true)
+    {
+    	viscosity_per_cell_temp = viscosity_per_cell;
+        typename DoFHandler<dim>::active_cell_iterator cell = dof_handler.begin_active();
+            for (unsigned int cellidx=0; cellidx<triangulation.n_active_cells(); ++cellidx, ++cell) // I am not using cellidx.. is that bad?
+            {
+            	for (int k=0;k<4;k++) // How do I remove this hard-coded 4?
+            	{
+            		if (cell->active())
+            		{
+						if(cell->neighbor_index(k)!=-1)
+						{
+							viscosity_per_cell = std::max(viscosity_per_cell_temp[cell->user_index()],viscosity_per_cell_temp[cell->neighbor(k)->user_index()]);
```

Comment:
Uhhh, viscosity_per_cell is a Vector and you are assigning to every single entry here? This is not what you want...